provider/activitypub/mastodon/client.go

```diff
@@ -1,67 +1,299 @@
 package mastodon
 
 import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
 	"fmt"
+	"io"
 	"net/http"
+	"net/url"
 	"time"
 
+	"github.com/go-fed/httpsig"
 	"github.com/go-playground/form/v4"
-	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/google/uuid"
+	"github.com/rss3-network/node/provider/activitypub"
+	"github.com/rss3-network/node/provider/httpx"
+	"go.uber.org/zap"
 )
 
 const (
-	DefaultTimeout  = 3 * time.Second
-	DefaultAttempts = 2
+	DefaultTimeout          = 2 * time.Second
+	DefaultAttempts         = 2
+	MessageChannelMaxLength = 1000
 )
 
 var _ Client = (*client)(nil)
 
 type Client interface {
-	GetKafkaConsumer() *kgo.Client
+	FollowRelayServices(ctx context.Context) error
+	GetMessageChan() (<-chan *map[string]interface{}, error)
+	GetActor() (*activitypub.Actor, error)
+	SendMessage(*map[string]interface{})
+	FetchAnnouncedObject(ctx context.Context, objectURL string) (*activitypub.Object, error)
+	// GetKafkaConsumer() *kgo.Client
 }
 
 type client struct {
-	httpClient    *http.Client
-	encoder       *form.Encoder
-	attempts      uint
-	kafkaConsumer *kgo.Client
+	httpClient *http.Client
+	encoder    *form.Encoder
+	attempts   uint
+	domain     string
+	privateKey *rsa.PrivateKey
+	signer     httpsig.Signer
+	msgChan    chan *map[string]interface{}
+	actor      *activitypub.Actor
+	relayURLs  []string
 }
 
```

Comment:
Consider adding comments to explain what each method does, especially for public methods.

Reply:
Resolved.